times/times.go

```diff
@@ -143,18 +142,16 @@ func getBootTimeUnixNano() int64 {
 		samples[i].t2 = time.Now()
 	}
 
-	sort.Slice(samples, func(i, j int) bool {
-		di := samples[i].t2.UnixNano() - samples[i].t1.UnixNano()
-		dj := samples[j].t2.UnixNano() - samples[j].t1.UnixNano()
-		if di < 0 {
-			di = -di
-		}
-		if dj < 0 {
-			dj = -dj
+	// Find the index with minimal time delta
+	md := int64(^uint64(0) >> 1) // same as math.MaxInt64
+	mi := 0
+	for i := range samples {
+		delta := samples[i].t2.UnixNano() - samples[i].t1.UnixNano()
```

Comment:
This is better in that we're now O(n) but we also changed the behavior and don't use absolute delta (to take into account edge cases where t2 < t1). The following maintains the previous behavior:

```suggestion
		delta := samples[i].t2.UnixNano() - samples[i].t1.UnixNano()
		if delta < 0 {
			delta = -delta
		}
```